config/src/lib.rs

```diff
@@ -158,6 +160,47 @@ impl ClientConfigBuilder {
     fn default_timeout(&self) -> Duration {
         Duration::from_secs(30)
     }
+
+    fn default_is_tls_enabled(&self) -> bool {
+        if let Some(controller_uri) = &self.controller_uri {
+            return match controller_uri.scheme() {
+                Ok(scheme) => scheme == TLS_SCHEME,
+                Err(_) => false,
+            };
+        }
+        false
+    }
+    /// validate the builder before returning it
+    ///
+    /// if both is_tls_enabled and controller_uri have been set
+    /// verify that the uri scheme (if present) matches the is_tls_enabled
+    /// value.
+    fn validate(&self) -> Result<(), String> {
+        if self.is_tls_enabled.is_none() || self.controller_uri.is_none() {
+            // at least one option has not been specified, cannot have a conflict
+            return Ok(());
+        }
+        let is_tls_enabled = self.is_tls_enabled.unwrap();
```

Comment:
Per the documentation:
> Default values are applied after validation, and will therefore not be validated!

So `is_tls_enabled` may be none, so unwrapping here is not a good idea. This should really use a `.unwrap_or(false)`

Reply:
unwrap cannot fail here due to the guard at line 179 `self.is_tls_enabled.is_none()`

I suppose a future change could remove that guard. I will apply the change you requested.